openmetadata-ui/src/main/resources/ui/src/components/common/Table/Table.tsx

```diff
@@ -206,11 +234,33 @@ const Table = <T extends Record<string, unknown>>(
     rest.staticVisibleColumns,
   ]);
 
+  useEffect(() => {
+    const storageKey = `selectedColumns-${currentUser?.fullyQualifiedName}`;
+    const selectedColumns = JSON.parse(
+      localStorage.getItem(storageKey) ?? '{}'
+    );
+
+    if (type || entityType) {
+      if (selectedColumns[type ?? entityType]) {
+        setColumnDropdownSelections(selectedColumns[type ?? entityType]);
+      } else if (defaultVisibleColumns) {
+        setColumnDropdownSelections(defaultVisibleColumns);
+        localStorage.setItem(
+          storageKey,
+          JSON.stringify({
+            ...selectedColumns,
+            [type ?? entityType]: defaultVisibleColumns,
+          })
+        );
+      }
+    }
+  }, [type, entityType, defaultVisibleColumns]);
```

Comment:
I think this can be a util and also let's do this based on some flag instead default